lib/wasix/src/fs/mod.rs

```diff
@@ -566,6 +566,11 @@ impl WasiFs {
         }
     }
 
+    pub fn clear_free_fd_list(&self) {
```

Comment:
Comment why this is needed.

Also, maybe call it  `unchecked_clear...` and add a `DANGEROUS` comment.